### PR TITLE
Bump version 1.7.0

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
- Bumps the version to 1.7.0
- Cleans up some release documentation that's out of date
- Bumps dependency versions for the website
- Updates our security policy to reference security@conjur.org.

### What ticket does this PR close?
n/a

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] This PR does not require updating any documentation, or
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs

#### (For releases only) Manual tests
- [x] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/providers/keychain)
- [x] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ debug.*
 .vscode/
 .idea
 .keep
+.jekyll-cache/
```
Comment (sgnn7) [inline .gitignore:8]: `j `before `k`

**Review by sgnn7 — COMMENTED** [review]
@izgeri 1 tiny nit

File: .gitignore (hunk 1 above)
Comment (izgeri) [reply .gitignore:8]: this is not at all alphabetical (see `vscode` just above)

Comment (sgnn7) [reply .gitignore:8]: Even if it isn't alphabetical, it seems like it's in the wrong section (it should be around other `.<dirname>/` entry which is `.vscode/`). Maybe we just need a quick cleanup of this file?

Comment (izgeri) [reply .gitignore:8]: ok, will take a stab at it

**Review by sgnn7 — APPROVED** [review]
@izgeri LGTM

**Pull request merged**